specs/GetFavicon.md

```diff
@@ -0,0 +1,155 @@
+                            m_appWindow->GetMainWindow(), WM_SETICON,
+                            ICON_SMALL, (LPARAM)icon);
+                    }
+                }
+
+                return S_OK;
+            }).Get());
+
+            return S_OK;
+    }).Get(), &m_faviconChangedToken));
+```
+## .NET / WinRT Registering a listener for favicon changes
+```c#
+webView.CoreWebView2.FaviconChanged += (CoreWebView2 sender, Object arg) =>
+{
+    System.IO.Stream stream = new System.IO.MemoryStream();
+    await webView.CoreWebView2.GetFaviconAsync(
+        CoreWebView2FaviconImageFormat.Png,
+        stream);
+    // setting the window Icon to the bitmap
+    this.Icon = BitmapFrame.Create(stream); 
+
+};
+```
+# API Notes
+Note that even if a web page does not have a Favicon, the FaviconChanged event
+is raised when the page is navigated to. The Favicon would be an
+empty image stream and empty Uri for the lack of a favicon. The end developer is expected to handle this scenario.
+Otherwise, we raise the FaviconChanged with an observed change to the
+Favicon. In that scenario, the CoreWebView2 has an updated value for the FaviconUri property, and the GetFavicon method to match the updated favicon.
+See [API Details](#api-details) Section below for API reference
+# API Details
+## Win32 C++
+```cpp
+/// This interface is a handler for when the `Favicon` is changed.
+/// The sender is the ICoreWebView2 object the top-level document of 
+/// which has changed favicon and the eventArgs is nullptr. Use the 
+/// FaviconUri property and GetFavicon method to obtain the favicon 
+/// data. The second argument is always null.
+/// For more information see `add_FaviconChanged`.
+[uuid(2913DA94-833D-4DE0-8DCA-900FC524A1A4), object, pointer_default(unique)]
+interface ICoreWebView2FaviconChangedEventHandler : IUnknown {
+  /// Called to notify the favicon changed.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] IUnknown* args);
+}
+
+/// This interface is a handler for the completion of the copying for the`imageStream`.
+/// The 'error_code` is E_NOT_SET if the there is no image. Otherwise error_code 
+/// is the result from the image write operation.
+/// For more details, see the `GetFavicon` API.
+[uuid(A2508329-7DA8-49D7-8C05-FA125E4AEE8D), object, pointer_default(unique)]
+interface ICoreWebView2GetFaviconCompletedHandler : IUnknown {
+  /// Called to notify the favicon has been retrieved.
+  HRESULT Invoke([in] HRESULT error_code);
+}
+
+/// This is the ICoreWebView2 Experimental Favicon interface.
+[uuid(DC838C64-F64B-4DC7-98EC-0992108E2157), object, pointer_default(unique)]
+interface ICoreWebView2_10 : ICoreWebView2_9 {
+    /// Add an event handler for the `FaviconChanged` event.
+    /// The `FaviconChanged` event is raised when the 
+    /// [favicon](https://developer.mozilla.org/en-US/docs/Glossary/Favicon)
+    /// of the top-level document changes or if script dynamically changes the favicon.
```

Comment:
I think this text is trying to distinguish between the favicon being set declaratively in the HTML versus script changing the favicon (although the script does so by dynamically modifying the HTML DOM's favicon element - maybe that's your point)

Reply:
There is a slight implied difference here. The implication is that if you navigate to a new page, the fav icon will fire a change event though it has "not technically changed".

Reply:
That's confusing me slightly.

It doesn't always raise the changed event when navigating to a new URI just it usually will since when navigating to a new URI we first don't have a favicon then we parse the document and may find a favicon right?

* On URI1 with favicon1
* Navigate to URI2
* Raise favicon changed event now that there is no favicon
* Parse the document and find a favicon
* Raise favicon changed event for favicon2


versus


* On URI1 with no favicon
* Navigate to URI2
* Don't raise favicon changed event because there continues to be no favicon
* Parse the document and find a favicon
* Raise favicon changed event for favicon1

Reply:
I adjusted the comment to match how the event is now being raised.